diff -Nuar eos-v1.6.0-org/scripts/eosio_build_ubuntu.sh eos-v1.6.0/scripts/eosio_build_ubuntu.sh
--- eos-v1.6.0-org/scripts/eosio_build_ubuntu.sh	2019-01-26 13:06:29.860738370 +0800
+++ eos-v1.6.0/scripts/eosio_build_ubuntu.sh	2019-01-26 13:10:15.791347825 +0800
@@ -24,8 +24,8 @@
 	printf "\\tDisk space total: %sG\\n" "${DISK_TOTAL%.*}"
 	printf "\\tDisk space available: %sG\\n" "${DISK_AVAIL%.*}"
 
-	if [ "${MEM_MEG}" -lt 7000 ]; then
-		printf "\\tYour system must have 7 or more Gigabytes of physical memory installed.\\n"
+	if [ "${MEM_MEG}" -lt 4096 ]; then
+		printf "\\tYour system must have 4 or more Gigabytes of physical memory installed.\\n"
 		printf "\\tExiting now.\\n"
 		exit 1
 	fi
